modules/wayfire/hm.nix

```diff
@@ -4,60 +4,67 @@
   pkgs,
   ...
 }:
+let
+  cfg = config.stylix.targets.wayfire;
+in
 
 {
-  options.stylix.targets.wayfire.enable =
-    config.lib.stylix.mkEnableTarget "wayfire" true;
+  options.stylix.targets.wayfire = {
+    enable = config.lib.stylix.mkEnableTarget "wayfire" true;
+    useWallpaper = config.lib.stylix.mkEnableWallpaper "wayfire" true;
+  };
 
-  config =
-    lib.mkIf (config.stylix.enable && config.stylix.targets.wayfire.enable)
-      (
-        let
-          inherit (config.lib.stylix) colors;
-          rgba = rgb: a: "\\#${rgb}${a}";
-          rgb = (lib.flip rgba) "ff";
+  config = lib.mkIf (config.stylix.enable && cfg.enable) (
+    let
+      inherit (config.lib.stylix) colors;
+      rgba = rgb: a: "\\#${rgb}${a}";
+      rgb = (lib.flip rgba) "ff";
 
-          wayfireConfig = config.wayland.windowManager.wayfire;
+      wayfireConfig = config.wayland.windowManager.wayfire;
 
-          wayfireBackground = pkgs.runCommand "wayfire-background.png" { } ''
+      wayfireBackground =
+        if cfg.useWallpaper then
+          (pkgs.runCommand "wayfire-background.png" { } ''
```

Comment:
> > [...] the recent /modules/wayfire/hm.nix module is not utilizing the useWallpaper option.
>
> should be fixed

Can this be simplified with `lib.mkIf`? Otherwise, simplify this if-then-else block with `lib.optionalString`.